camel-core/src/main/java/org/apache/camel/component/seda/SedaProducer.java

```diff
@@ -236,13 +238,20 @@ protected void addToQueue(Exchange exchange, boolean copy) throws SedaConsumerNo
         }
 
         log.trace("Adding Exchange to queue: {}", target);
-        if (blockWhenFull) {
+        if (blockWhenFull && offerTimeout == 0) {
             try {
                 queue.put(target);
             } catch (InterruptedException e) {
                 // ignore
                 log.debug("Put interrupted, are we stopping? {}", isStopping() || isStopped());
             }
+        } else if(blockWhenFull && offerTimeout > 0) {
+            try {
+                queue.offer(target, offerTimeout, TimeUnit.MILLISECONDS);
```

Comment:
I would've expected the result of this call to be checked. And on "false" throw an IllegalStateException. Otherwise there is no way for the caller to know if the message was successfully added.

Reply:
Done . please review now